tests/Unit/Evolution/Systems/RelativisticEuler/Valencia/Test_FixConservatives.cpp

```diff
@@ -0,0 +1,101 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <algorithm>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/RelativisticEuler/Valencia/FixConservatives.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "tests/Unit/TestCreation.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+
+template <size_t Dim>
+void test_fix_conservatives(
+    const RelativisticEuler::Valencia::FixConservatives<Dim>& variable_fixer) {
+  Scalar<DataVector> tilde_d{DataVector{1.2, 9.e-13, 2.0, 0.0, 8.7}};
+  Scalar<DataVector> tilde_tau{DataVector{-0.001, 0.7, 0.1, 5.5, -0.05}};
+  auto tilde_s =
+      make_with_value<tnsr::i<DataVector, Dim, Frame::Inertial>>(tilde_d, 0.0);
+  for (size_t i = 0; i < Dim; ++i) {
+    tilde_s.get(i) = DataVector{1.25, -1.56, 4.32, 5.0, 0.0};
+  }
+  // Use identity matrix as metric for simplicity
+  auto sqrt_det_spatial_metric =
```

Comment:
I think it makes more sense to specify the inverse spatial metric and compute the determinant from it than the other way around.